From 8b7e5d278c3e0008de31b61f6d50893260967755 Mon Sep 17 00:00:00 2001
From: zhang xingxing <2388969553@qq.com>
Date: Thu, 15 Oct 2026 04:01:57 +0200
Subject: [PATCH] avformat/idroqdec: skip the whole body of a repeated RoQ_INFO
 chunk

The RoQ player in the Quake III Arena source (code/client/cl_cin.c) also
continues after the declared size of every RoQ_INFO chunk.

Fixes: yyLBi9GsHV9w
Fixes: swarm-RoQ_INFO-re-entry-skips-fixed-size-ignoring-chunk_size-e7709a
Stream desynchronization Replicated through UnModified FFmpeg

Found-by: zhang xingxing <2388969553@qq.com>
Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>
(cherry picked from commit 8b7e5d278c3e0008de31b61f6d50893260967755)
---
 libavformat/idroqdec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/idroqdec.c b/libavformat/idroqdec.c
index 861338ceec3..76d96196bac 100644
--- a/libavformat/idroqdec.c
+++ b/libavformat/idroqdec.c
@@ -142,7 +142,7 @@ static int roq_read_packet(AVFormatContext *s,
                 break;
             }
             /* don't care about this chunk anymore */
-            avio_skip(pb, RoQ_CHUNK_PREAMBLE_SIZE);
+            avio_skip(pb, chunk_size);
             break;
 
         case RoQ_QUAD_CODEBOOK:
-- 
2.53.0

